libs/hdf-converters/src/checklist-mapper.ts

```diff
@@ -0,0 +1,454 @@
+        _.set(
+          vuln,
+          'severityoverride',
+          data.stigs[stig_index].vulns[vuln_index].severityOverride
+        );
+        _.set(
+          vuln,
+          'severityjustification',
+          data.stigs[stig_index].vulns[vuln_index].severityJustification
+        );
+      });
+    });
+
+    return revertChecklist(data.raw);
+  }
+}
+
+const IMPACT_MAPPING: Map<string, number> = new Map([
+  ['high', 0.7],
+  ['medium', 0.5],
+  ['low', 0.3],
+  ['none', 0.0]
+]);
+
+/**
+ * Tranformer function that splits a string and return array
+ * @param input - string of CCI references
+ * @returns ref - array of CCI references
+ */
+function cciRef(input: string): string[] {
+  return input.split('; ');
+}
+
+const CCI_NIST_MAPPING = new CciNistMapping();
+
+/**
+ * Transformer function that splits string and maps resulting array
+ * into NIST control tags
+ * @param input - string of CCI references
+ * @returns tag - array of NIST Control Tags
+ */
+function nistTag(input: string): string[] {
+  const identifiers: string[] = input.split('; ');
+  return CCI_NIST_MAPPING.nistFilter(
+    identifiers,
+    DEFAULT_STATIC_CODE_ANALYSIS_NIST_TAGS
+  );
+}
+
+/**
+ * Transformer function that checks if the status is 'Not Applicable' returning a 0.
+ * Otherwise, maps severity to IMPACT_MAPPING
+ * @param vuln - checklist vulnerability object
+ * @returns impact - number 0.0, 0.3, 0.5, or 0.7
+ * @default is 0
+ */
+function transformImpact(vuln: Record<string, any>): number {
+  if (vuln.status === 'Not Applicable') return 0.0;
+  const severity = findSeverity(vuln);
+  return IMPACT_MAPPING.get(severity.toString().toLowerCase()) || 0;
+}
+
+/**
+ * Inner function to check is there was a severify override which would alter
+ * the impact of the vulnerability
+ * @param vuln - checklist vulnerability object
+ * @returns - severity
+ */
+function findSeverity(vuln: Record<string, any>): string {
+  if (vuln.severityOverride.length > 0) {
+    return vuln.severityOverride;
+  }
+  return vuln.severity;
+}
+
+/**
+ * Transformer function that uses current heimdall checklist export syntax for
+ * findingDetails attribute to separate a single string into multiple
+ * result objects
+ * @param input - array of one element consisting of {code_desc, status, start_time}
+ * @returns ExecJSON.ControlResult
+ */
+function parseFindingDetails(
+  input: Record<string, any>
+): ExecJSON.ControlResult[] {
+  const results: ExecJSON.ControlResult[] = [];
+  const findings: string = input[0].code_desc;
+  if (findings.length !== 0) {
+    const splitFindings: string[] = findings.split(
+      '--------------------------------\n' // join variable identified from cklResults function in ExportCKLModal.vue
+    );
+    splitFindings.forEach((details) => {
+      let codeDesc = '';
+      let status = '';
+      let message = '';
+      const splitResults: string[] = details.split(/\n(.*)/s, 2); // split using the first new line character only
```

Comment:
splitresults[0] = status
splitresults[1] = code_desc

Reply:
Updated to make code easier to read and spits into usable variables instead of accessing via index.